src/NosCore.Controllers/ExchangePacketController.cs

```diff
@@ -0,0 +1,251 @@
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using NosCore.Configuration;
+using NosCore.GameObject;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.ComponentEntities.Interfaces;
+using NosCore.GameObject.Networking;
+using NosCore.GameObject.Services.ExchangeInfo;
+using NosCore.GameObject.Services.ItemBuilder;
+using NosCore.GameObject.Services.ItemBuilder.Item;
+using NosCore.Packets.ClientPackets;
+using NosCore.Packets.ServerPackets;
+using NosCore.Shared.Enumerations;
+using NosCore.Shared.Enumerations.Buff;
+using NosCore.Shared.Enumerations.Character;
+using NosCore.Shared.Enumerations.Interaction;
+using NosCore.Shared.Enumerations.Items;
+using NosCore.Shared.I18N;
+using Serilog;
+
+namespace NosCore.Controllers
+{
+    public class ExchangePacketController : PacketController
+    {
+        private readonly WorldConfiguration _worldConfiguration;
+        private readonly ExchangeAccessService _exchangeAccessService;
+        private readonly ILogger _logger = Logger.GetLoggerConfiguration().CreateLogger();
+        
+        public ExchangePacketController(WorldConfiguration worldConfiguration, ExchangeAccessService exchangeAccessService)
+        {
+            _worldConfiguration = worldConfiguration;
+            _exchangeAccessService = exchangeAccessService;
+        }
+
+        [UsedImplicitly]
+        public ExchangePacketController()
+        {
+
+        }
+
+        [UsedImplicitly]
+        public void ExchangeList(ExcListPacket packet)
+        {
+            if (packet.Gold > Session.Character.Gold || packet.BankGold > Session.Account.BankMoney)
+            {
+                return;
+            }
+
+            var list = new ServerExcListPacket
+            {
+                SenderType = SenderType.Server,
+                VisualId = Session.Character.VisualId,
+                Gold = packet.Gold,
+                BankGold = packet.BankGold,
+                SubPackets = new List<ServerExcListSubPacket>()
+            };
+            
+            var target = Broadcaster.Instance.GetCharacter(s => s.VisualId == Session.Character.ExchangeData.TargetVisualId && s.MapInstanceId == Session.Character.MapInstanceId) as Character;
+
+
+            if (packet.SubPackets.Any())
+            {
+                byte i = 0;
+                foreach (var value in packet.SubPackets)
+                {
+                    if (value.PocketType == PocketType.Bazaar)
```

Comment:
better remove this and delete the pocketType bazaar it won't be implemented as the opennos one